lib/tools/gitignoreScanner.js

```diff
@@ -0,0 +1,35 @@
+// Module for scanning the .gitignore and constructing a list of files to ignore
+const fs = require('fs')
+const path = require('path')
+
+module.exports = function (gitignorePath) {
+  let fileList = ['lib-cov', 'node_modules', 'pids', 'logs', 'results', 'Thumbs.db', '*.seed', 'npm-debug.log', '.build', 'public', '*.log', '*.csv', '*.dat', '*.out', '*.pid', '*.gz', '.DS_Store']
+  let gitData
+  let polishedList = []
+
+  try {
+    // ensure that we can access .gitignore
+    fs.accessSync(gitignorePath)
+  } catch (e) {
+    // if we can not, return list of defaults
+    return fileList
+  }
+
+  // get data from .gitignore, trim it of whitespace, split at new lines
+  gitData = fs.readFileSync(gitignorePath).toString().trim().split('\n')
+
+  // skip lines already in fileList as well as blank lines/comments
+  gitData.forEach((line) => {
```

Comment:
Trim in individual line in this loop before examining the contents to defend against random whitespace.